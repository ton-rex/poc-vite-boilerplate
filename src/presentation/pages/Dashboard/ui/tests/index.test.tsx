import { describe, vi } from 'vitest'
import { testSnapshots } from '~/infra/utils/tests'
import * as states from './componentStates'

vi.mock('@stone-payments/jade', () => {
  return {
    Button: ({ children: text }: { children: string }) => {
      return <button>{text}</button>
    },
    Text: ({ children: text }: { children: string }) => {
      return <p>{text}</p>
    },
    Tag: ({ children: text }: { children: string }) => {
      return <p>{text}</p>
    },
  }
})

describe('DashboardContainer', (): void => {
  testSnapshots(states)
})
