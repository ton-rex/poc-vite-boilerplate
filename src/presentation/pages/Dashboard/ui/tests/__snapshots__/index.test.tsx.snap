// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DashboardContainer > renders 'default' state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Dashboard
        </p>
        <p>
          Multimarca
        </p>
        <p>
          Mulplataforma
        </p>
        <p>
          Multitema
        </p>
        <button
          class="sc-bdnyFh jgfhu"
        >
          Multitema
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Dashboard
      </p>
      <p>
        Multimarca
      </p>
      <p>
        Mulplataforma
      </p>
      <p>
        Multitema
      </p>
      <button
        class="sc-bdnyFh jgfhu"
      >
        Multitema
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DashboardContainer > renders 'generic error' state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Dashboard
        </p>
        <p>
          Multimarca
        </p>
        <p>
          Mulplataforma
        </p>
        <p>
          Multitema
        </p>
        <button
          class="sc-bdnyFh jgfhu"
        >
          Multitema
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Dashboard
      </p>
      <p>
        Multimarca
      </p>
      <p>
        Mulplataforma
      </p>
      <p>
        Multitema
      </p>
      <button
        class="sc-bdnyFh jgfhu"
      >
        Multitema
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DashboardContainer > renders 'loading' state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Dashboard
        </p>
        <p>
          Multimarca
        </p>
        <p>
          Mulplataforma
        </p>
        <p>
          Multitema
        </p>
        <button
          class="sc-bdnyFh jgfhu"
        >
          Multitema
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Dashboard
      </p>
      <p>
        Multimarca
      </p>
      <p>
        Mulplataforma
      </p>
      <p>
        Multitema
      </p>
      <button
        class="sc-bdnyFh jgfhu"
      >
        Multitema
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`DashboardContainer > renders 'network error' state correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <p>
          Dashboard
        </p>
        <p>
          Multimarca
        </p>
        <p>
          Mulplataforma
        </p>
        <p>
          Multitema
        </p>
        <button
          class="sc-bdnyFh jgfhu"
        >
          Multitema
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <p>
        Dashboard
      </p>
      <p>
        Multimarca
      </p>
      <p>
        Mulplataforma
      </p>
      <p>
        Multitema
      </p>
      <button
        class="sc-bdnyFh jgfhu"
      >
        Multitema
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
